src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -209,7 +209,13 @@ def compile(
         with self.monkeypatch.context() as m:
             m.setenv(inmanta_lsm.const.ENV_INSTANCE_ID, str(service_id))
             m.setenv(inmanta_lsm.const.ENV_INSTANCE_VERSION, str(service.version))
-            m.setenv(inmanta_lsm.const.ENV_PARTIAL_COMPILE, str(self.partial_compile))
+
+            try:
+                m.setenv(inmanta_lsm.const.ENV_PARTIAL_COMPILE, str(self.partial_compile))
+            except AttributeError:
+                # This attribute only exists for iso5+, iso4 doesn't support partial compile.
+                # We then simply don't set the value.
+                pass
```

Comment:
Shouldn't we raise an exception if it's set to `True`?

Reply:
I think an exception would be a bit too much, a warning would be enough?

Reply:
Fine by me